src/horus_api.c

```diff
@@ -98,19 +138,42 @@ struct horus *horus_open (int mode, int Rs) {
         hstates->rx_bits_len = hstates->max_packet_len;
     }
 
-    if (mode == HORUS_MODE_BINARY) {
+    if (mode == HORUS_MODE_BINARY_V1) {
+        // Parameter setup for the Legacy Horus Binary Mode (22 byte frames, Golay encoding)
+
         hstates->mFSK = 4;
-        hstates->max_packet_len = HORUS_BINARY_NUM_BITS;
-        for (i=0; i<sizeof(uw_horus_binary); i++) {
-            hstates->uw[i] = 2*uw_horus_binary[i] - 1;
+        hstates->max_packet_len = HORUS_BINARY_V1_NUM_CODED_BITS;
+
+        // If baud rate not provided, use default
+        if (hstates->Rs == -1){
+            hstates->Rs = HORUS_BINARY_V1_DEFAULT_BAUD;
+        }
+
+        if (tx_tone_spacing == -1){
+            // No tone spacing provided. Disable mask estimation, and use the default tone spacing value as a dummy value.
+            tx_tone_spacing = HORUS_BINARY_V1_DEFAULT_TONE_SPACING;
+            mask = 0;
+        } else {
+            // Tone spacing provided, enable mask estimation.
+            mask = 1;
         }
-        hstates->uw_len = sizeof(uw_horus_binary);
-        hstates->uw_thresh = sizeof(uw_horus_binary) - 2; /* allow a few bit errors in UW detection */
+
+        for (i=0; i<sizeof(uw_horus_binary_v1); i++) {
+            hstates->uw[i] = 2*uw_horus_binary_v1[i] - 1;
+        }
+        hstates->uw_len = sizeof(uw_horus_binary_v1);
+        hstates->uw_thresh = sizeof(uw_horus_binary_v1) - 2; /* allow a few bit errors in UW detection */
         horus_l2_init();
         hstates->rx_bits_len = hstates->max_packet_len;
     }
-   
-    hstates->fsk = fsk_create(hstates->Fs, hstates->Rs, hstates->mFSK, 1000, 2*hstates->Rs);
+    // TODO: Horus 256/128-bit modes here.
+
```

Comment:
OK this PR is a work in progress.  So shall we just keep it open until the lockdown project (new protocols) is finished?  I'm no interested in merging incomplete work.  You can still run it out of this branch if you need to.

Reply:
The aim of this PR was to try and bring codec2 and FreeDV into some sort of alignment so that they both build correctly, while giving some of the improvements of the updated FSK modem. I decided to try and bring some further improvements at the same time.

Reply:
Given that we're not using the mask estimator for V1 (since I guess the Eb/N0 area the mask estimator brings improvements in doesn't help when using a golay code) I could just change all the freedv changes to just call horus_open with horus_open(HORUS_MODE_BINARY,100), and be done with it...

However, I figured it would be useful to try and get the horus_api into the expected state that it would be in once these new modes are developed, so that some of the extra functionality that might be helpful in freedv (selection of horus mode, input of tone spacing to help mask estimator, display of frequency estimator outputs) could be developed in parallel, and tested using the V1 modem, or even with RTTY decoding.

I could also just remove all the references to the V2 modems. This PR would still provide an improvement.